scripts/zkai_stwo_statement_envelope_benchmark.py

```
@@ -0,0 +1,717 @@
+
+
+def _check_artifact_bindings(envelope: dict[str, Any], proof: dict[str, Any]) -> None:
+    artifacts = _required_dict(envelope.get("artifacts"), "artifacts")
+    program_path = _repo_path(_required_artifact_reference(artifacts, "program_path"))
+    metadata_path = _artifact_dir_path(_required_artifact_reference(artifacts, "metadata_path"))
+    _required_artifact_reference(artifacts, "proof_path")
+    metadata = _required_dict(_load_json(metadata_path), "metadata")
+    statement = _required_dict(envelope.get("statement"), "statement")
+    expected_program_hash = metadata.get("artifacts", {}).get(STWO_PROGRAM_PATH)
+    if sha256_file(program_path) != expected_program_hash:
+        raise StwoEnvelopeError("program artifact hash does not match metadata")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)